main.cpp

```diff
@@ -0,0 +1,197 @@
+        if (pixel_color[0] != color[0] || pixel_color[1] != color[1] || pixel_color[2] != color[2]) {
+            return lines;
+        }
+    }
+
+    // 如果所有像素的颜色都与特定的RGB颜色相同，则绘制蓝色的直线
+    lines.emplace_back(point1, point2);
+    return lines;
+}
+
+int matrix_size, num_lines;
+vector<int> path;
+vector<vector<int>> edges;
+vector<vector<bool>> visited_edge;
+vector<int> target_path;
+
+bool backtrack(int u) {
+    if (path.size() == num_lines + 1) {
+        target_path = path;
+        return true;
+    }
+    bool flag = false;
+    for (int i = 0; i < matrix_size; i++) {
+        if (edges[u][i] && !visited_edge[u][i] && !visited_edge[i][u]) {
+            path.push_back(i);
+            visited_edge[u][i] = true;
+            visited_edge[i][u] = true;
+            if (backtrack(i)) flag = true;
+            visited_edge[u][i] = false;
+            visited_edge[i][u] = false;
+            path.pop_back();
+        }
+    }
+    return flag;
+}
+
+void add(int a, int b) {
+    edges[a][b] = 1;
+    edges[b][a] = 1;
+}
+
+int main() {
+    char img_path[80];
+    cin >> img_path;
+    Mat img = cv::imread(img_path);
+    int loc_max = 0;
+    vector<Point> points;
+    std::vector<std::pair<cv::Point, cv::Point>> lines;
+
+    Mat image = imread(img_path, IMREAD_GRAYSCALE);
+    imshow("raw", image);
+    Mat kernel = getStructuringElement(MORPH_RECT, Size(3, 3));
+    Mat dilate_img;
+    dilate(image, dilate_img, kernel);
+    Mat erode_img;
+    erode(image, erode_img, kernel);
+
+    Mat absdiff_img;
+    absdiff(dilate_img, erode_img, absdiff_img);
+    Mat threshold_img;
+    threshold(absdiff_img, threshold_img, 40, 255, THRESH_BINARY);
+    Mat result;
+    bitwise_not(threshold_img, result);
+//    imshow("circles", result);
+
+    vector<Vec3f> circles;
+    HoughCircles(result, circles, HOUGH_GRADIENT, 1, 10, 100, 15, 15, 30);
+    // 绘制圆形包围框
+    for (auto &i: circles) {
+        // 圆的坐标
+        int x = i[0];
+        int y = i[1];
+        int r = i[2];
+        // 绘制圆形包围框
+        if (x > loc_max) {
+            loc_max = x;
+        }
+        if (y > loc_max) {
+            loc_max = y;
+        }
+        int times = ceil(loc_max / 50);
+        for (int tmp = 0; tmp < times + 1; tmp++) {
+            if (x > 50 * tmp - 5 && x < 50 * tmp + 5) {
+                x = 50 * tmp;
+            }
+            if (y > 50 * tmp - 5 && y < 50 * tmp + 5) {
+                y = 50 * tmp;
+            }
+        }
+        points.emplace_back(x, y);
+        circle(img, Point(x, y), r + 2, Scalar(211, 204, 255), -1);
+        circle(img, Point(x, y), 2, Scalar(255, 255, 255), -1);
+    }
+    //查询circles的个数
+    int num_circles = circles.size();
+    cout << num_circles << endl;
+
+    matrix_size = num_circles;
+
+    edges.resize(matrix_size);
+    visited_edge.resize(matrix_size);
+    for (int i = 0; i < matrix_size; i++) {
+        edges[i].resize(matrix_size);
+        visited_edge[i].resize(matrix_size);
+    }
```

Comment:
使用多重resize让人难以理解，为什么不直接在这里重新给edges和visited_edge赋值呢呢  
```suggestion
    edges = vector<vector<int>>(num_circles, vector<int>(num_circles, 0));
    visited_edge = vector<vector<bool>>(num_circles, vector<int>(num_circles, false));
```

Reply:
已解决